core/client/src/main/java/alluxio/client/lineage/AbstractLineageClient.java

```diff
@@ -63,12 +65,19 @@ public long createLineage(List<AlluxioURI> inputFiles, List<AlluxioURI> outputFi
     // TODO(yupeng): relax this to support other type of jobs
     Preconditions.checkState(job instanceof CommandLineJob,
         PreconditionMessage.COMMAND_LINE_LINEAGE_ONLY);
-    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    LineageMasterClient masterClient = null;
     try {
+      masterClient = mContext.acquireMasterClient();
       long lineageId = masterClient.createLineage(stripURIList(inputFiles),
           stripURIList(outputFiles), (CommandLineJob) job);
       LOG.info("Created lineage {}", lineageId);
       return lineageId;
+    } catch (NotFoundException e) {
+      throw new FileDoesNotExistException(e.getMessage());
+    } catch (UnavailableException e) {
+      throw new IOException(e.getMessage());
+    } catch (AlluxioStatusException e) {
+      throw e.toAlluxioException();
```

Comment:
There should be a consistent pattern:
```java
try {
  operation()
} catch (SpecificStatusExceptionCorrespondingToThrownSubclassOfAlluxioExceptionForThisSpecificMethod e) {
  throw new ThrownSubclassOfAlluxioExceptionForThisSpecificMethod1(e.getMessage());
...
} catch (UnavailableException e) {
  throw e.toIOException();
} catch (AlluxioStatusException e) {
  throw e.toAlluxioException();
}
```

For example, 

```java
  @Override
  public boolean deleteLineage(long lineageId, DeleteLineageOptions options)
      throws IOException, LineageDoesNotExistException, LineageDeletionException, AlluxioException {
    LineageMasterClient masterClient = null;
    try {
      masterClient = mContext.acquireMasterClient();
      boolean result = masterClient.deleteLineage(lineageId, options.isCascade());
      LOG.info("{} delete lineage {}", result ? "Succeeded to " : "Failed to ", lineageId);
      return result;
    } catch (NotFoundException e) {
      throw new LineageDoesNotExistException(e.getMessage());
    } catch (FailedPreconditionException e) {
      throw new LineageDeletionException(e.getMessage());
    } catch (UnavailableException e) {
      throw e.toIOException();
    } catch (AlluxioStatusException e) {
      throw e.toAlluxioException();
    } finally {
      mContext.releaseMasterClient(masterClient);
    }
  }
```

I forgot to handle `UnavailableException` in `BaseFileSystem`, it's updated now.

After 2.0 all of this will go away and we won't catch anything  at the public API boundary